### DISCUSSION TITLE
CenterPopup 组件 --max-width 无法更改

### DISCUSSION THREAD

**Discussion by JasonZhongZexin** [ant-design/ant-design-mobile#6575] [Q&A]
在使用CenterPopup时，组件默认的--max-width 是75vw，根据文档描述--max-width在css中是可以自行修改的。实际使用过程中，无论在css或者typescript中设置新的--max-width值后发现这个值并没作用到组件中。

Typescript
```
<CenterPopup className={styles.centerPopup} destroyOnClose={true}
        style={{'--max-width':'95vw'}}
        bodyClassName={styles.centerPopup}
        visible={loginPopupVisible}
        onMaskClick={handleLoginMaskClick}>
        <div><PopupLogin onClose={handleLoginMaskClick} showRegister={showRegisterPopup} showRestPass={showForgetPassPopup} /></div>
</CenterPopup>
```

CSS
```
.centerPopup{
  background-color:transparent;
  --adm-center-popup-background-color:transparent;
  --max-width:'95vw';
}
```
上面例子中无论是在css中尝试更改 --max-width  还是在css中更改，当在浏览器查看的时候，centerPopup的最大宽度仍旧是75vw

以下为更改后在浏览器里看到的css的样式：
```
.adm-center-popup {
    --background-color: var(--adm-center-popup-background-color, var(--adm-color-background));
    --border-radius: var(--adm-center-popup-border-radius, 8px);
    --max-width: var(--adm-center-popup-max-width, 75vw);
    --min-width: var(--adm-center-popup-min-width, 280px);
    --z-index: var(--adm-center-popup-z-index, 1000);
    position: fixed;
    z-index: var(--z-index);
}
```

**Comment by zombieJ** [comment]
参考一下文档，`--min-width` 和 `--max-width` 作用在 `.adm-center-popup-wrap` 上，所以你需要给 `.adm-center-popup-wrap` 设置变量参数就会有效果了：

<img width="950" alt="截屏2024-03-19 14 03 21" src="https://github.com/ant-design/ant-design-mobile/assets/5378891/da17c22d-2ce7-4701-b44b-f9c86af6c668">

或者就是参考文档提供的全局变量参数 `--adm-center-popup-max-width` 和 `--adm-center-popup-min-width` 设置：

<img width="952" alt="截屏2024-03-19 14 04 53" src="https://github.com/ant-design/ant-design-mobile/assets/5378891/5ac574b4-5976-42fe-ad3f-5ecbc692ae12">
